code/cvss/src/Security/CVSS.hs

```diff
@@ -107,16 +107,15 @@ data Metric = Metric
 -- | Parse a CVSS string.
 parseCVSS :: Text -> Either CVSSError CVSS
 parseCVSS txt
-  | "CVSS:3.1/" `Text.isPrefixOf` txt = CVSS CVSS31 <$> validateComponents validateCvss31
-  | "CVSS:3.0/" `Text.isPrefixOf` txt = CVSS CVSS30 <$> validateComponents validateCvss30
-  | "CVSS:2.0/" `Text.isPrefixOf` txt = CVSS CVSS20 <$> validateComponents validateCvss20
-  | otherwise = Left UnknownVersion
+  | "CVSS:3.1/" `Text.isPrefixOf` txt = CVSS CVSS31 <$> validateComponents True validateCvss31
+  | "CVSS:3.0/" `Text.isPrefixOf` txt = CVSS CVSS30 <$> validateComponents True validateCvss30
```

Comment:
We probably ought to have catch for unknown versions, before the `otherwise` (which infers CVSS v2.0):
```haskell
  | "CVSS:" `Text.isPrefixOf` txt = Left UnknownVersion
```

Reply:
Ok, but the `otherwise` branch assumes there is no version prefix, so a v4 string still produces an error:

```haskell
λ> parseCVSS "CVSS:4.0/AV:L/AC:L/AT:P/PR:L/UI:N/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N"
Left Missing value for "CVSS:4.0"
```